types/genesis.go

```diff
@@ -113,7 +120,7 @@ func GenesisDocFromJSON(jsonBlob []byte) (*GenesisDoc, error) {
 	genDoc := GenesisDoc{}
 	err := cmtjson.Unmarshal(jsonBlob, &genDoc)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("invalid json for GenesisDoc: %s", err)
```

Comment:
Good catch, making errors clear are good. I wonder if a message like "failed to unmarshall" would be better.

Reply:
I think writing errors in the form "failed to do X" or "error doing X" is a bit redundant. It's an error, so we already know that something failed to do something 😄 

When writing errors, I try as much as possible to follow the advice [here](https://google.github.io/styleguide/go/best-practices#adding-information-to-errors), because I think it leads to better, more useful errors.

But if there is an error style we use in the codebase, I'll follow it. 

Any thoughts?

Reply:
Good point.